GPU/GPUTracking/TPCClusterFinder/GPUTPCNNClusterizerHost.cxx

```diff
@@ -12,21 +12,99 @@
 /// \file GPUTPCNNClusterizerHost.cxx
 /// \author Christian Sonnabend
 
+#include <CommonUtils/StringUtils.h>
+
 #include "GPUTPCNNClusterizerHost.h"
 #include "GPUTPCNNClusterizer.h"
+#include "CCDB/CcdbApi.h"
 #include "GPUSettings.h"
 #include "ML/3rdparty/GPUORTFloat16.h"
 
 using namespace o2::gpu;
 
-GPUTPCNNClusterizerHost::GPUTPCNNClusterizerHost(const GPUSettingsProcessingNNclusterizer& settings, GPUTPCNNClusterizer& clusterer)
+GPUTPCNNClusterizerHost::GPUTPCNNClusterizerHost(const GPUSettingsProcessingNNclusterizer& settings, int32_t deviceId)
+{
+  init(settings, deviceId);
+}
+
+void GPUTPCNNClusterizerHost::loadFromCCDB(std::map<std::string, std::string> settings)
+{
+  o2::ccdb::CcdbApi ccdbApi;
+  ccdbApi.init(settings["nnCCDBURL"]);
+
+  metadata["inputDType"] = settings["inputDType"];
+  metadata["outputDType"] = settings["outputDType"];
+  metadata["nnCCDBEvalType"] = settings["nnCCDBEvalType"];         // classification_1C, classification_2C, regression_1C, regression_2C
+  metadata["nnCCDBWithMomentum"] = settings["nnCCDBWithMomentum"]; // 0, 1 -> Only for regression model
+  metadata["nnCCDBLayerType"] = settings["nnCCDBLayerType"];       // FC, CNN
+  if (settings["nnCCDBInteractionRate"] != "" && std::stoi(settings["nnCCDBInteractionRate"]) > 0) {
+    metadata["nnCCDBInteractionRate"] = settings["nnCCDBInteractionRate"];
+  }
+  if (settings["nnCCDBBeamType"] != "") {
+    metadata["nnCCDBBeamType"] = settings["nnCCDBBeamType"];
+  }
+
+  bool retrieveSuccess = ccdbApi.retrieveBlob(settings["nnCCDBPath"], ".", metadata, 1, false, settings["outputFile"]);
```

Comment:
please don't use the ccdb api directly in a workflow, data should come from the DPL workflow, which has to subscribe to the CCDB.

Reply:
For now I would like to be able to fetch networks from the test ccdb, as I don't have push / delete rights on the actual ccdb. For production I will change it to use the dpi workflow api. Do you maybe have a place (file) in mind where I could do the fetching or maybe how I can pass a reference to the API to the GPUTPCNNClusterizerHost::loadFromCCDB as input?

Reply:
I would not do the ccdb ccess inside the GPUTracking library.
The CCDB access should happen inside the GPUWorkflowSpec, and then we should pass in the objects probbly as calibration objects.
Then you can have an option for the GPUWorkflowSpec to either request the objects from the ccdb or for testing purpises get them from the testccdb.
There is a class GPUCalibIbjects or so, if you want you can add a pointer to NeuralNetworks.
Normally, these calib objects are then supposed to be made available on tge GPU as well, but obviously this is not needed for NNs, since that is then handled by ONNX. So basically, if you add one pointer that will just become one 8byte unused bogus pointer in const memory, so will not hurt much, and can bebimproved later.